kedro/extras/datasets/spark/spark_dataset.py

```diff
@@ -309,9 +317,32 @@ def __init__(  # pylint: disable=too-many-arguments
         if save_args is not None:
             self._save_args.update(save_args)
 
+        # Handle schema
+        self._schema = self._load_schema(self._load_args.pop("schema_json_path", None))
         self._file_format = file_format
         self._fs_prefix = fs_prefix
 
+    @staticmethod
+    def _load_schema(schema_json_path: str) -> Optional[StructType]:
+        if schema_json_path is None:
+            return None
+
+        # TODO Limit protocols to file only?
+        # TODO What about files in HDFS?
+        # TODO What about credentials, e.g., schema stored in separate GCS bucket?
+        protocol, schema_path = get_protocol_and_path(schema_json_path)
+        file_system = fsspec.filesystem(protocol)
+        pure_posix_path = PurePosixPath(schema_path)
+        load_path = get_filepath_str(pure_posix_path, protocol)
+
+        # Open schema file
+        with file_system.open(load_path) as fs_file:
+
+            # TODO lazy load schema when loading dataframe?
```

Comment:
I was thinking of "lazy loading" in the sense of loading it when the `_load` function of the dataset was being called. Do you think this would make sense?